influxql/select.go

```diff
@@ -460,6 +460,11 @@ func buildExprIterator(expr Expr, ic IteratorCreator, opt IteratorOptions, selec
 			return itr, nil
 		}
 	case *BinaryExpr:
+		// If there was no fill, force null fill to ensure LHS and RHS stay in sync.
+		// (opt is a value, not a reference, so modifications should not affect callers' scope.)
+		if opt.Fill == NoFill {
```

Comment:
I think this is the wrong place as it will affect all queries and make it impossible to use `fill(none)` with any binary expression. But, not having it here will result in non-CQ queries not being aligned correctly too. I think we need to match the times in the binary expression iterators since, otherwise, this will end up acting as a regression.

If you want to force `NullFill` for CQ's only, you can do that in `influxql/iterator.go`.